ecmascript/utils/src/lib.rs

```diff
@@ -99,6 +99,49 @@ impl Visit for IdentFinder<'_> {
     }
 }
 
+pub fn contains_arguments<N>(body: &N) -> bool
+where
+    N: VisitWith<ArgumentsFinder>,
+{
+    let mut visitor = ArgumentsFinder { found: false };
+    body.visit_with(&Invalid { span: DUMMY_SP } as _, &mut visitor);
+    visitor.found
+}
+
+pub struct ArgumentsFinder {
+    found: bool,
+}
+
+impl Visit for ArgumentsFinder {
+    noop_visit_type!();
+
+    /// Don't recurse into constructor
+    fn visit_constructor(&mut self, _: &Constructor, _: &dyn Node) {}
+
+    /// Don't recurse into fn
+    fn visit_function(&mut self, _: &Function, _: &dyn Node) {}
+
+    /// Don't recurse into member expression prop if not computed
+    fn visit_member_expr(&mut self, m: &MemberExpr, _: &dyn Node) {
+        m.obj.visit_with(m, self);
+        match &*m.prop {
+            Expr::Ident(_) if !m.computed => {}
+            _ => m.prop.visit_with(m, self),
+        }
+    }
+
```

Comment:
Can you add `visit_prop` which checks for `{ arguments }`?
`Prop::Shorthand(Ident)` may contain arguments, too.